graphql/types.lua

```diff
@@ -317,8 +317,27 @@ types.long = types.scalar({
   isValueOfTheType = isLong,
 })
 
+-- Return `false` for NaN, Negative Infinity or Positive Infinity.
+-- Return `true` otherwise.
+local function isfinite(n)
+  local d = n - n
+  return n == n and d == d
+end
+
 local function isFloat(value)
-  return type(value) == 'number'
+  -- http://spec.graphql.org/October2021/#sec-Float
+  --
+  -- > Non-finite floating-point internal values (NaN and Infinity) cannot be
+  -- > coerced to Float and must raise a field error.
+  if type(value) == 'number' then
+      return isfinite(value)
```

Comment:
Indent seems to be broken here. Below you use 2 spaces.

Reply:
Overlooked it, thanks! Fixed.